### ISSUE TITLE
Large CSV download succeeds, but never prompts me for where to save the file??

### ISSUE THREAD

**Issue by rjmackay** [ushahidi/platform#2217]
### Expected behaviour
CSV download works and asks me where to save the file

### Actual behaviour
CSV download runs (HTTP request succeeds!) but the file is never saved

### Steps to reproduce the behaviour/error
- Go to data view
- Select Share
- Pick export to CSV

##### Where
- [ ] Other (explain): Uchaguzi Sandbox

**Comment by rowasc** [comment]
@rjmackay this is the browser pretty much crashing with those large files that are not being downloaded (we get it as normal content, not with the correct headers, then we manipulate it, etc) . When the number of posts is enough that you can still get them but it doesn't prompt for download, we have verified that you can get them with postman/curl/whatever because they don't crash like the browser on that response.  
TLDR:  the browser thinks that the download is still in progress

**Comment by rjmackay** [comment]
Ugh. Everything is terrible!
Do we just need to fix the headers then? or something else?

**Comment by rowasc** [comment]
Yes, that's an accurate description. :(
So, we need a few things. 
- we need to fix the headers to do a proper download, and related to that: 
- -  we need to STOP manipulating our CSV in the client (!)
- -  we need to have a way to access those CSVs ( a list) because otherwise the user will be super confused and they will lose their CSV processing request when they leave a page. 
- - we need some way (queues, we think) to scale how we do CSVs on the backend , and obviously to improve the sorting which is terrible 
-  in the backend area, we need to fix CSV performance in general. Will moved things to a console command, which helped somewhat because at least you can get them offline if you are in need (as a cli user.. ugh) but  it still ends up running out of memory with very large datasets

**Comment by rowasc** [comment]
https://github.com/ushahidi/platform/issues/2121
https://github.com/ushahidi/platform/issues/2109
Related issues we have been creating...

**Comment by jshorland** [comment]
@rowasc i'm thinking about increasing this to p0. it completely stops the user from being able to download to CSV, correct?

**Comment by jshorland** [comment]
Yeah this is definitely a thing. Making this a p0. I just continue to get "download failed -
 network error" when trying to download ALL the data out of sandbox.uchaguzi.or.ke. Not sure if related, but there's no way for me to download large data sets.

**Comment by rjmackay** [comment]
Seems to be working in sandbox

**Comment by jshorland** [comment]
UPDATE: this is NOT working in sandbox anymore, nor in uchaguzi.

**Comment by rowasc** [comment]
@jshorland update as discussed the error we get is from the backend timeout , which is terrible =(  (just updating to avoid confusion internally). Moving discussion to platform#2154 to keep things together.
